core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -450,6 +446,12 @@ protected boolean authenticate(String userName, Object credential, boolean domai
             }
         }
 
+        if (log.isDebugEnabled()) {
```

Comment:
This code will work. But ideally the if conditions should be swapped. First check whether authenticated. Then check for debug enabled and log